options/ansi/generic/file-io.cpp

```diff
@@ -331,6 +335,23 @@ int abstract_file::_write_back() {
 	return 0;
 }
 
+int abstract_file::_discard_buffer() {
+	if (_type == stream_type::file_like) {
+		// save our current location
+		off_t new_offset;
+		auto seek_offset = (off_t(__offset) - off_t(__io_offset));
+		if (int e = io_seek(seek_offset, SEEK_CUR, &new_offset); e) {
```

Comment:
Why do we need to seek in the underlying FD?

Reply:
Because as far as I saw there was now way to save the current position in the file properly, without rewriting some of the other code. I could probably rewrite it to avoid seeking but I was reluctant to do that.

Reply:
As discussed over Discord, this seek makes sense. Maybe add a comment that we seek to reset the position of the FD to the position of the FILE.